### PR TITLE
meta: automate npm releases

### PR DESCRIPTION
**Pull request by thomasheartman**
This PR does this:

It adds auto-release-to-npm functionality for tags that start with `v`.

It also updates the version of some of the actions we run to align with the unleash/unleash repo.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - 'v*'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
```
Comment (chriswk) [inline .github/workflows/release.yml:14]: Do we actually need a matrix, if we're only publishing for one node-version?

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -20,9 +20,9 @@ jobs:
         # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v2
+      uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node-version }}
```
Comment (chriswk) [inline .github/workflows/node.js.yml:27]: Maybe also worth it to add caching?
https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

**Review by chriswk — APPROVED** [review]
Some optimisation suggestions, but great to have one less manual job for releasing :)

File: .github/workflows/release.yml (hunk 1 above)
Comment (chriswk) [reply .github/workflows/release.yml:14]: Same thing here, setup-node supports adding cache for yarn

Comment (thomasheartman) [reply .github/workflows/release.yml:14]: Don't know about the matrix, but a single column, single row matrix is still just one element, right? I suppose the fix would be to remove the `strategy` property and instead use this in the step:

```yml
with:
  node-version: '14'
```

But if we do want to expand the matrix for later, then I don't mind leaving it in. Also provides an easy place to look for versions, I think 💁🏼

As for the caching, yeah, I'll add it!

**Pull request merged**